code/part-two/processor/test/06-AddResponse.js

```diff
@@ -0,0 +1,248 @@
+
+describe('Create Offer', function() {
+  let handler = null;
+  let context = null;
+  let offer = null;
+  let ownerPub = null;
+  let ownerPriv = null;
+  let responderPub = null;
+  let responderPriv = null;
+  let moji = null;
+
+  before(function() {
+    handler = new MojiHandler();
+  });
+
+  beforeEach(function() {
+    context = new Context();
+    const ownerTxn = new Txn({ action: 'CREATE_COLLECTION' });
+    const responderTxn = new Txn({ action: 'CREATE_COLLECTION' });
+
+    ownerPub = ownerTxn._publicKey;
+    ownerPriv = ownerTxn._privateKey;
+    responderPub = responderTxn._publicKey;
+    responderPriv = responderTxn._privateKey;
+
+    return handler.apply(responderTxn, context)
+      .then(() => {
+        const address = getAddress.collection(responderPub);
+        moji = decode(context._state[address]).moji.sort();
+        return handler.apply(ownerTxn, context);
+      })
+      .then(() => {
+        const ownerAddress = getAddress.collection(ownerPub);
+        const { moji } = decode(context._state[ownerAddress]);
+        offer = getAddress.offer(ownerTxn._publicKey)(moji);
+
+        const offerTxn = new Txn({ action: 'CREATE_OFFER', moji }, ownerPriv);
+        return handler.apply(offerTxn, context);
+      });
+  });
+
+  it('should add a response with multiple moji', function() {
+    const txn = new Txn({ action: 'ADD_RESPONSE', moji, offer }, responderPriv);
+
+    return handler.apply(txn, context)
+      .then(() => {
+        const { responses } = decode(context._state[offer]);
+        expect(responses, 'Offer should have one response').to.have.lengthOf(1);
+
+        expect(responses[0].approver, 'Response should need owner approval')
+          .to.equal(ownerPub);
+        expect(responses[0].moji, 'Response should include the moji offered')
+          .to.deep.equal(moji);
+      });
+  });
+
+  it('should add a response with a single moji', function() {
+    moji = moji.slice(0, 1);
+    const txn = new Txn({ action: 'ADD_RESPONSE', moji, offer }, responderPriv);
+
+    return handler.apply(txn, context)
+      .then(() => {
+        const responses = decode(context._state[offer]).responses;
+        expect(responses, 'Offer should have one response').to.have.lengthOf(1);
+
+        expect(responses[0].approver, 'Response should need owner approval')
+          .to.equal(ownerPub);
+        expect(responses[0].moji, 'Response should include the moji offered')
+          .to.deep.equal(moji);
+      });
+  });
+
+  it('should add a response from the offer owner', function() {
+    const txn = new Txn({ action: 'ADD_RESPONSE', moji, offer }, ownerPriv);
+
+    return handler.apply(txn, context)
+      .then(() => {
+        const { responses } = decode(context._state[offer]);
+        expect(responses[0].approver, 'Response should need responder approval')
+          .to.equal(responderPub);
+      });
+  });
+
+  it('should sort the moji in the response', function() {
```

Comment:
is the ln 94 test of `should sort the moji in the response` necessary if we have the ln 106 test of `should reject responses identical to one previously added`?

Reply:
Yeah, I think we need it. You could check if pass the test on 106 without sorting the moji you store. I will add this test to CREATE_OFFER as well.